nifi-nar-bundles/nifi-standard-bundle/nifi-standard-processors/src/main/java/org/apache/nifi/security/util/crypto/PasswordBasedEncryptor.java

```diff
@@ -121,8 +122,8 @@ public void process(final InputStream in, final OutputStream out) throws IOExcep
             byte[] salt;
             try {
                 // NiFi legacy code determined the salt length based on the cipher block size
-                if (cipherProvider instanceof NiFiLegacyCipherProvider) {
-                    salt = ((NiFiLegacyCipherProvider) cipherProvider).readSalt(encryptionMethod, in);
+                if (cipherProvider instanceof org.apache.nifi.security.util.crypto.NiFiLegacyCipherProvider) {
```

Comment:
This pattern is repeated a few times, is there a way to reduce the repetition?

Reply:
chatted with andy and he taught me about deprecation handling.  importing the class would have caused a deprecation warning.  doing the long form usage of the class each time is the way to explicitly reference/intentionally, a deprecated class.